### PR TITLE
Rule disallow `{{unless}}`

### LINKED ISSUE #30: Rule: Disallow `{{unless}}`
Good:

``` hbs
{{#unless somePredicate}}
{{/unless}}

<div class="{{unless foo 'no-foo'}}"></div>
```

Bad:

``` hbs
{{#unless (and foo bar)}}
{{/unless}}
```

``` hbs
{{#unless foo}}
{{else}}
{{/unless}}
```

``` hbs
{{#if foo}}
{{else unless bar}}
{{/if}}
```

Mostly because it is impossible to reason about....

### PR DESCRIPTION
**Pull request by duggiefresh**
Fixes #30

### CODE REVIEW

**Comment by duggiefresh** [comment]
I'll update the README as well. Just wanted to get some 👀 on this beforehand. Thanks!

File: lib/rules/lint-unless-helper.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+const buildPlugin = require('./base');
+
+const messages = {
+  followingElseBlock: 'Invoking an `{{else}}` helper with the `unless` helper is not recommended',
```
Comment (rwjblue) [inline lib/rules/lint-unless-helper.js]: Small verbiage tweak here:

> Using an `{{else}}` block with `{{unless}}` should be avoided.

File: lib/rules/lint-unless-helper.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  followingElseIfBlock: 'Invoking an `{{else if}}` helper with the `unless` helper is not recommended',
```
Comment (rwjblue) [inline lib/rules/lint-unless-helper.js]: I think we can use the `followingElseBlock` for this one (e.g. we don't need two different messages in this case).

File: lib/rules/lint-unless-helper.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  asElseUnlessBlock: 'Invoking an `{{else unless}}` helper is not recommended',
+  withHelper: 'Invoking an `{{unless}}` helper with another template helper is not recommended'
```
Comment (rwjblue) [inline lib/rules/lint-unless-helper.js]: Small verbiage tweak here:

> Using `{{unless}}` in combination with other helpers should be avoided.

File: lib/rules/lint-unless-helper.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+};
+
+function logMessage(context, message, loc, actual) {
```
Comment (rwjblue) [inline lib/rules/lint-unless-helper.js]: If you put this on your `class` below, you can call it via `this.logMessage()` and not pass in `context`.  I don't have a significant preference, just curious if you do 😄...

**Comment by duggiefresh** [comment]
Thanks for the review @rwjblue! Updated per request, and I added a basic entry in the README

**Comment by rwjblue** [comment]
OK, I rebased over the bonkers changes that I have been doing on master, and updated this to account for those tweaks.

Thank you @duggiefresh!!!

**Pull request merged**